### PR TITLE
Add native VibeVoice ASR support

### LINKED ISSUE #81: Support vibevoice_asr (audio diffusion pipeline)
## Overview

`vibevoice_asr` is Microsoft's speech recognition/synthesis model using a 4-stage diffusion pipeline. Available models: `microsoft/VibeVoice-ASR` (512K downloads), `microsoft/VibeVoice-1.5B` (63K downloads).

**model_type:** `vibevoice`  
**HF archs:** `VibeVoiceForASRTraining`, `VibeVoiceForConditionalGeneration`

## Architecture (4 stages)

```
Audio input
  → Semantic Tokenizer  (causal EnCodec-style CNN, ratios=[8,5,5,4,2,2])
  → Acoustic Tokenizer  (separate causal CNN encoder, same structure)
  → Qwen2 LM Decoder    (hidden=1536–3584, 28 layers — reusable)
  → DDPM Diffusion Head (20 inference steps, cosine schedule, latent_size=64)
  → Audio output
```

| Component | Type | Reusable? |
|-----------|------|-----------|
| Semantic tokenizer | Causal CNN (EnCodec-style) | ❌ New |
| Acoustic tokenizer | Causal CNN (EnCodec-style) | ❌ New |
| Qwen2 LM decoder | Standard transformer | ✅ Reuse Qwen2CausalLMModel |
| DDPM diffusion head | Diffusion (20 steps, cosine) | ❌ New |

## Complexity: High

This is **not a standard transformer model** — it is a diffusion-based audio pipeline. New infrastructure required:

1. **Causal convolutional audio encoder** (EnCodec-style, depthwise conv with downsampling ratios)
2. **VQ-VAE / audio tokenizer** for semantic and acoustic token quantization
3. **DDPM diffusion head** — `vibepod_diffusion_head`, cosine noise schedule, 20-step inference loop
4. **Multi-stage pipeline** — unlike standard `input_ids → logits`, this has 4 distinct stages with different I/O contracts

Estimated effort: **4–6 weeks** (new component family for audio diffusion).

## Dependencies

- Audio diffusion infrastructure (partially exists for image diffusion — see diffusion models skill)
- Causal CNN encoder component (new, similar to audio codec encoders)
- DDPM inference loop support in ONNX graph
- New `AudioDiffusionTask` or similar task class

## Priority: P2

Lower download counts relative to complexity. The Qwen2 backbone is reusable, but represents only a small fraction of the total work. Recommend deferring until audio diffusion infrastructure is established (potentially after image diffusion is more mature).

## Related

- `microsoft/VibeVoice-Realtime-0.5B` (364K downloads) — same architecture, smaller
- `microsoft/VibeVoice-ASR-HF` (257K downloads) — HF-compatible variant
- Issue #80 (protein/biology models) — independent

### PR DESCRIPTION
**Pull request by justinchuby**
Adds staged offline ASR support exclusively for `microsoft/VibeVoice-ASR-HF@f22241c2062b3b25272bf117397e03d73381037a` (`vibevoice_asr` / `VibeVoiceAsrForConditionalGeneration`). The legacy `microsoft/VibeVoice-ASR` training checkpoint is explicitly rejected before download; it is not aliased.\n\nThe five-stage package routes all 901 native checkpoint tensors with no ignored weights. Processor/tokenizer assets and runtime metadata use the native HF repository. L4/L5 real-weight transcription remains unverified due to GPU capacity.

### CODE REVIEW

File: docs/vibevoice-asr.md
Comment (justinchuby) [inline docs/vibevoice-asr.md:1]: Move this to model docstring so it shows up in docs site

Comment (justinchuby) [reply docs/vibevoice-asr.md:1]: Moved and consolidated the ASR architecture/runtime contract into `VibeVoiceASRForConditionalGeneration`'s class docstring, which the generated model documentation publishes. Removed the duplicate standalone page and updated the catalog entry.

File: tests/weight_alignment_test.py [hunk 1]
```diff
@@ -170,6 +170,89 @@ def test_vibevoice_native_hf_weights_cover_every_stage_parameter():
...
+
+    from mobius.models.vibevoice_asr import VibeVoiceASRForConditionalGeneration
+    from mobius.models.vibevoice_test import _make_tiny_hf_config
+
+    index_path = hf_hub_download(
+        "microsoft/VibeVoice-ASR",
+        filename="model.safetensors.index.json",
+        revision="d0c9efdb8d614685062c04425d91e01b6f37d944",
+        local_dir=tmp_path,
+    )
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply tests/weight_alignment_test.py:191]: Changed the pinned-index download to use `cache_dir=tmp_path`. `cache_dir` is supported by the project's older Hugging Face Hub compatibility range, while `local_dir` is not.
